composer/core/state.py

```diff
@@ -482,6 +506,9 @@ def load_state_dict(self, state: Dict[str, Any], strict: bool = False):
             if attribute_name == 'model':
                 self.load_model_state(state, strict=strict)
                 continue
+            elif attribute_name == 'dataloader_state':
```

Comment:
Following up on the above comment, could we do something like this (to avoid the extra layer of wrapping)?

```suggestion
            elif attribute_name == 'dataloader':
                 load_dataloader_state(self.dataloader, state)
```

Reply:
The serialization logic will need a similar check.